src/Tools/Source/CompilerGeneratorTools/Source/IOperationGenerator/IOperationClassWriter.cs

```diff
@@ -0,0 +1,305 @@
+                    {
+                        WriteLine("// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.");
+
+                        WriteUsing("System");
+
+                        if (node.Properties.Any(f => IsImmutableArray(f.Type)))
+                        {
+                            WriteUsing("System.Collections.Immutable");
+                        }
+
+                        if (grouping.Key == "FlowAnalysis")
+                        {
+                            WriteUsing("Microsoft.CodeAnalysis.Operations");
+                        }
+
+                        Blank();
+
+                        WriteStartNamespace(grouping.Key ?? "Operations");
+                        WriteInterface(node);
+                        WriteEndNamespace();
+                        _writer.Flush();
+                    }
+                }
+            }
+
+            WriteOperationKind();
+        }
+
+        private void WriteUsing(string nsName)
+        {
+            WriteLine($"using {nsName};");
+        }
+
+        private void WriteStartNamespace(string? namespaceSuffix)
+        {
+            WriteLine($"namespace Microsoft.CodeAnalysis{(namespaceSuffix is null ? "" : $".{namespaceSuffix}")}");
+            Brace();
+        }
+
+        private void WriteEndNamespace()
+        {
+            Unbrace();
+        }
+
+        private void WriteInterface(AbstractNode node)
+        {
+            WriteComments(node.Comments, writeReservedRemark: true);
+
+            if (node.Obsolete is { } obsolete)
+            {
+                WriteLine($"[Obsolete({obsolete.Message}, error: {obsolete.ErrorText})]");
+            }
+
+            WriteLine($"{(node.IsInternal ? "internal" : "public")} interface {node.Name} : {node.Base}");
+            Brace();
+
+            foreach (var property in node.Properties)
+            {
+                WriteInterfaceProperty(property);
+            }
+
+            Unbrace();
+        }
+
+        private void WriteComments(Comments? comments, bool writeReservedRemark)
+        {
+            if (comments is object)
+            {
+                bool hasWrittenRemarks = false;
+
+                foreach (var el in comments.Elements)
+                {
+                    WriteLine($"/// <{el.LocalName}>");
+
+                    string[] separators = new[] { "\r", "\n", "\r\n" };
+                    string[] lines = el.InnerXml.Split(separators, StringSplitOptions.RemoveEmptyEntries);
+
+                    int indentation = lines[0].Length - lines[0].TrimStart().Length;
+
+                    foreach (var line in lines)
+                    {
+                        if (string.IsNullOrWhiteSpace(line)) continue;
+                        WriteLine($"/// {line.Substring(indentation)}");
```

Comment:
Thank you for using interpolated strings in this generator, the other ones are borderline unreadable without it.